build.assets/tooling/cmd/check/main.go

```diff
@@ -108,7 +110,28 @@ func checkLatest(ctx context.Context, tag string, gh github.GitHub) error {
 	return nil
 }
 
+// checkValidSemver returns an error if the supplied string is not a valid
+// Semver identifier
+func checkValidSemver(tag string) error {
+	if !semver.IsValid(tag) {
+		return trace.BadParameter("version is invalid semver: %v", tag)
+	}
+	return nil
+}
+
 func checkPrerelease(tag string) error {
+	// While an invalid semver string is not *technically* the same thing
+	// as having pre- or post-release tags, both of the tests below return
```

Comment:
I don't think there is any such thing as a "post-release tag". I find this "prerelease" check confusing enough, given that the tool returns false when run with `-check prerelease` and the tag IS a prerelease tag. What I think this function really is is checking if a tag is a valid *release* tag and the check you are adding here solidifies that.

I feel it would make more sense if this comment documented this reverse-case test and how an invalid server fits into that.

> checkPrerelease returns nil if the given tag is not a pre-release tag and is also a valid tag. An error is returned for tags that are not a valid semver string or if they contain a pre-prelease or build component.

With that comment, I think this whole comment block is not needed.

Ideally, we'd rename this function to `checkRelease` and also change the flag value to `release` or similar. That would require changing all callers, so perhaps just add `release` as an alias for `prerelease` and when the callers of this get moved to GitHub Actions, it can use the `release` flag value then?

Reply:
Yeah, I'm using the wrong terminology for semver. What I was thinking of as "post-release" is actually the build metadata tag - the one with the leading `+`. I think I internalised it as leading `-` means _pre-release_ so a leading `+` means _post release_.

Reply:
Anyway, I have switched the internal logic so that it makes a bit more sense (I hope). I've left the UI as-is for now, but def agree that calling it a `prerelease` check is confusing.